src/transaction.rs

```diff
@@ -0,0 +1,40 @@
+use serde::Deserialize;
+use serde::Serialize;
+
+pub type Address = u32;
+pub type Amount = u16;
+pub type Nonce = u32;
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Hash, Clone, Debug)]
+pub struct Enter {
+    // TODO: Address should be assigned automatically by the rollup.
+    pub account: Address,
+    pub amount: Amount,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Hash, Clone, Debug)]
+pub struct Transfer {
+    pub from: Address,
+    pub to: Address,
+    pub amount: Amount,
+    pub nonce: Nonce,
+}
+
+#[derive(Serialize, Deserialize, PartialEq, Eq, Hash, Clone, Debug)]
+pub enum Transaction {
+    Enter(Enter),
+    Transfer(Transfer),
+}
```

Comment:
Apparently, you cannot use enum variants as standalone types. What I really wanted is this:
```rust
pub enum Transaction {
     Enter { account: Address, amount: Amount},
     Transfer { from: Address, to: Address, amount: Amount, nonce: u32 },
}
```

but the problem is that you cannot create instances of `Transaction::Transfer` and pass them around without having to pattern match at every step. This was especially annoying in combination with `thiserror` crate, because I cannot pattern match when accessing attributes of an error.

Having separate structs is a passable workaround but it unfortunately makes some uses more verbose. For example matching against a specific variant requires repeating the name:
```rust
Transaction::Enter(Enter { account, amount }) => ...
```